setup.py

```diff
@@ -89,30 +50,68 @@ def yajl_present():
         os.remove(yajl_version_test_file.name)
 
 
-def patch_yajl_sources():
-    """Make yajl sources ready for direct compilation against them"""
+ORIGINAL_SOURCES = os.path.join("cextern", "yajl")
+PATCHED_SOURCES = "yajl"
+
+
+def patch_yajl_sources() -> None:
+    """Make yajl sources ready for direct compilation against them."""
     # cp cextern/yajl -R $yajl_sources_copy
     # mkdir $yajl_sources_copy/yajl
     # cp $yajl_sources_copy/src/api/*.h $yajl_sources_copy/yajl
-    patched_sources = os.path.join(tempfile.mkdtemp(), 'yajl')
-    shutil.copytree(os.path.join('cextern', 'yajl'), patched_sources)
-    headers_original = os.path.join(patched_sources, 'src', 'api')
-    headers_copy = os.path.join(patched_sources, 'yajl')
+    if os.path.isdir(PATCHED_SOURCES):
+        print("Sources are already in the correct place. Skip copy")
+        return
+    shutil.copytree(
+        ORIGINAL_SOURCES, PATCHED_SOURCES,
+        ignore=shutil.ignore_patterns("example", "test"),
+    )
+    headers_original = os.path.join(PATCHED_SOURCES, "src", "api")
+    headers_copy = os.path.join(PATCHED_SOURCES, "yajl")
     shutil.copytree(headers_original, headers_copy)
-    return patched_sources
 
 
+def cleanup_yajl_sources() -> None:
+    """Cleanup yajl sources."""
+    if os.path.isdir(PATCHED_SOURCES):
+        shutil.rmtree(PATCHED_SOURCES)
+
+
+class CustomSDist(sdist):
+    def run(self) -> None:
+        self.execute(patch_yajl_sources, ())
+        sdist.run(self)
+        self.execute(cleanup_yajl_sources, ())
+
+
+class CustomBuildExt(build_ext):
+    def run(self) -> None:
+        self.execute(patch_yajl_sources, ())
+        build_ext.run(self)
+        self.execute(cleanup_yajl_sources, ())
+
+
+cmdclass = {}
 extra_sources = []
 extra_include_dirs = []
 libs = ['yajl']
 embed_yajl = os.environ.get('IJSON_EMBED_YAJL', None) == '1'
 if not embed_yajl:
     have_yajl = yajl_present()
 else:
-    yajl_sources = patch_yajl_sources()
+    cmdclass = {"sdist": CustomSDist, "build_ext": CustomBuildExt}
+    if not os.path.isdir(PATCHED_SOURCES):
+        # sdist: First pass in the source tree.
+        # Files haven't been copied yet.
+        yajl_sources = ORIGINAL_SOURCES
+    else:
+        # wheel: Second pass in the isolated build environment.
+        # Only copied files are available.
+        yajl_sources = PATCHED_SOURCES
```

Comment:
The fact you are doing this seems a bit confusing because you _are_ copying the files on `sdist`, as well as on `build_ext`, so in both cases you can safely refer to `PATCHED_SOURCES`.

But then again see my other comment about reworking this whole set of changes to bring things back to being simpler